src/types/routeWithParams.ts

```diff
@@ -1,48 +1,10 @@
-import { ExtractParamName } from '@/types/params'
-import { LiteralParam, Param, ParamGetSet, ParamGetter } from '@/types/paramTypes'
+import { ExtractRouteParamTypesWithOptional } from '@/types/params'
 import { Routes } from '@/types/route'
 import { RoutesName, RoutesMap } from '@/types/routesMap'
-import { Identity } from '@/types/utilities'
-import { MakeOptional } from '@/utilities/makeOptional'
-import type { ZodSchema } from 'zod'
 
 export type RouteGetByKey<TRoutes extends Routes, TKey extends RoutesName<TRoutes>> = RoutesMap<TRoutes>[TKey]
 
 export type RouteParamsByKey<
   TRoutes extends Routes,
   TKey extends string
-> = ExtractRouteParamTypesWithoutLosingOptional<RouteGetByKey<TRoutes, TKey>>
-
-type ExtractRouteParamTypesWithoutLosingOptional<TRoute> = TRoute extends {
-  host: { params: infer HostParams extends Record<string, Param> },
-  path: { params: infer PathParams extends Record<string, Param> },
-  query: { params: infer QueryParams extends Record<string, Param> },
-  hash: { params: infer HashParams extends Record<string, Param> },
-}
-  ? ExtractParamTypesWithoutLosingOptional<HostParams & PathParams & QueryParams & HashParams>
-  : Record<string, unknown>
```

Comment:
@pleek91 not saying it's correct but here's where that came from

Reply:
gotcha, so it just moved and the diff just makes it look like a change 👍